frontend/src/components/datasets/ShareDatasetModal.tsx

```diff
@@ -46,9 +46,12 @@ export default function ShareDatasetModal(props: ShareDatasetModalProps) {
 	const [options, setOptions] = useState([]);
 	const users = useSelector((state: RootState) => state.group.users);
 
-	const setUserRole = (datasetId: string, username: string, role: string) =>
+	const setUserRole = async (datasetId: string, username: string, role: string) =>
```

Comment:
oh why is the `async` needed

Reply:
To make sure that when the subsequent listRoles is called, the update is already reflected in the backend

Reply:
ah sorry i didn't realize you are directly using dispatch :-) My bad it makes perfect sense now.